scripts/mc_log_autocorrelations.py

```diff
@@ -9,78 +9,142 @@
-    # redis server also include timestamps (as JDs), but I think it's actually
-    # preferable to use our own clock here. Note that we also ensure that the
-    # records grabbed in one execution of this script have identical
-    # timestamps, which is a nice property.
-    time = datetime.datetime.utcnow()
-
-    for ant in antnums:
-        for pol in 'xy':
-            try:
-                data = rsession.hgetall('visdata://%d/%d/%s%s' % (ant, ant, pol, pol))
-            except Exception as e:
-                print('failed to get autocorrelation %d%s: %s (%s)'
-                      % (ant, pol, e, e.__class__.__name__),
-                      file=sys.stderr)
-                n_failures += 1
-                continue
-
-            data = data.get('data')
-            if data is None:
-                print('failed to get autocorrelation %d%s: no "data" item?' % (ant, pol),
-                      file=sys.stderr)
-                n_failures += 1
-                continue
-
-            data = np.fromstring(data, dtype=np.float32)
-
-            # For now, we just compute the median:
-
-            ac = autocorrelations.Autocorrelations()
-            ac.time = time
-            ac.antnum = ant
-            ac.polarization = pol
-            ac.measurement_type = autocorrelations.MeasurementTypes.median
-            ac.value = np.asscalar(np.median(data))  # must turn np.float32 into plain Python float
-
-            if args.debug:
-                print(data.shape, repr(ac))
-            if not args.noop:
-                dbsession.add(ac)
-
-if n_failures:
-    print('error: %d fetches failed' % n_failures, file=sys.stderr)
-    sys.exit(1)
+    try:
+        while True:
+
+            rsession = test_redis_connection(rsession, redis_pool)
+
+            hostname = socket.gethostname()
+            keys = [
+                k.decode("utf-8")
+                for k in rsession.keys()
+                if k.startswith(b"auto") and not k.endswith(b"timestamp")
+            ]
+
+            ants = []
+            pols = []
+            for key in keys:
+                match = re.search(r"auto:(?P<ant>\d+)(?P<pol>e|n)", key)
+                if match is not None:
+                    ant, pol = int(match.group("ant")), match.group("pol")
+                    ants.append(ant)
+                    pols.append(pol)
+            ants = np.unique(ants)
+            pols = np.unique(pols)
+
+            now = Time.now()
+            auto_time = Time(
+                np.frombuffer(rsession.get("auto:timestamp"), dtype=np.float64).item(),
+                format="jd",
+            )
+
+            if abs(now - auto_time) >= TimeDelta(300, format='sec'):
+                print(
+                    "Autocorrelations are at least 5 minutes old, "
+                    "or from the future, adding values as None."
+                )
+                autos_none = True
+            else:
+                autos_none = False
```

Comment:
This looks like you're going to try to add `None` to the DB as the values. That will fail because the columns are not nullable.

Reply:
Good catch. I think we decided it _should_ be nullable though right? So I'll just allow `None` for the value column in the autocorrelations?

Reply:
why add them at all if they're none? It looks like you're if there's no new data, you're going to add a whole bunch of rows with Nones. Why not just leave them out and check the timestamp?

Reply:
Okay. No need to flood the database with Nones.